pkg/retriever/assignablecandidatefinder.go

```diff
@@ -3,54 +3,72 @@ package retriever
 import (
 	"context"
 	"fmt"
+	"sync/atomic"
 	"time"
 
+	"github.com/benbjohnson/clock"
 	"github.com/filecoin-project/lassie/pkg/events"
+	"github.com/filecoin-project/lassie/pkg/internal/candidatebuffer"
 	"github.com/filecoin-project/lassie/pkg/types"
 )
 
 // AssignableCandidateFinder finds and filters candidates for a given retrieval
 type AssignableCandidateFinder struct {
 	isAcceptableStorageProvider IsAcceptableStorageProvider
 	candidateFinder             CandidateFinder
+	clock                       clock.Clock
 }
 
+const BufferWindow = 5 * time.Millisecond
+
 func NewAssignableCandidateFinder(candidateFinder CandidateFinder, isAcceptableStorageProvider IsAcceptableStorageProvider) AssignableCandidateFinder {
-	return AssignableCandidateFinder{candidateFinder: candidateFinder, isAcceptableStorageProvider: isAcceptableStorageProvider}
+	return NewAssignableCandidateFinderWithClock(candidateFinder, isAcceptableStorageProvider, clock.New())
+}
+func NewAssignableCandidateFinderWithClock(candidateFinder CandidateFinder, isAcceptableStorageProvider IsAcceptableStorageProvider, clock clock.Clock) AssignableCandidateFinder {
+	return AssignableCandidateFinder{candidateFinder: candidateFinder, isAcceptableStorageProvider: isAcceptableStorageProvider, clock: clock}
 }
-func (rcf AssignableCandidateFinder) FindCandidates(ctx context.Context, request types.RetrievalRequest, eventsCallback func(types.RetrievalEvent)) ([]types.RetrievalCandidate, error) {
+func (rcf AssignableCandidateFinder) FindCandidates(ctx context.Context, request types.RetrievalRequest, eventsCallback func(types.RetrievalEvent), onCandidates func([]types.RetrievalCandidate)) error {
```

Comment:
we have the acronym mismatch thing here now too, `rcf` maybe should be `acf`? not critical

Reply:
from when it was the RetrievalCandidateFinder, needs to be changed.